pkg/proctree/process.go

```diff
@@ -0,0 +1,383 @@
+package proctree
+
+import (
+	"sync"
+	"time"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/aquasecurity/tracee/pkg/utils/types"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// processNode is a node in the process tree representing a process.
+// Its purpose is to keep the information of the process, and the connections to other processes
+// and threads nodes.
+// The current implementation is using mutex, so all internal members are assumed to be protected
+// by it (so are not necessarily thread-safe types)
+type processNode struct {
+	// TODO: Add information about the processes like opened files,
+	//  network activities (like TCP connections), argv, environment variables,
+	//  loader and interpreter
+	id             int
```

Comment:
is this the PID (process id) or some other id ? maybe rename it to "pid" ?

Reply:
So it is the PID, as PID stands for Process ID and this is a process node.
@yanivagman do you think this should be changed to `pid`?